docs/modules/integrate/pages/integrate-with-langchain-java.adoc

```diff
@@ -0,0 +1,266 @@
+= Integrate with LangChain for Java
+:description: The Hazelcast integration for LangChain provides a vector store implementation that enables using Hazecast Vector Search with the LangChain4J Java framework.
+
+{description}
+
+== Introduction
+
+link:https://docs.langchain4j.dev[LangChain4J] is a Java framework that makes it easier to create large language model (LLM) based solutions, such as chat bots, by linking various components.
+
+LangChain4J's `EmbeddingStore` interface makes it easier to incorporate RAGs (Retrieval Augmented Generation) in LLM solutions.
+
+The `hazelcast.com:langchain-hazelcast` package provides the Hazelcast `EmbeddingStore` implementation for LangChain.
+
+== Installing the LangChain/Hazelcast embedding store
+
+To install the `hazelcast.com:langchain-hazelcast` package, add the following lines to your `pom.xml` file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.hazelcast</groupId>
+    <artifactId>langchain-hazelcast</artifactId>
+    <version>6.0.0</version>
+</dependency>
+----
+
+== Creating an embedding store
+
+The Hazelcast embedding store is implemented using the `HazelcastEmbeddingStore`class  in the `hazelcast.com:langchain-hazelcast` package.
+
+Before creating the embedding store, you must create an instance of the embedding model. The model instance will be used to generate the embeddings for adding text documents and searching them.
+
+The following example uses `AllMiniLmL6V2QuantizedEmbeddingModel`, but you can use any model:
+
+[source,java]
+----
+var embeddingModel = new AllMiniLmL6V2QuantizedEmbeddingModel();
+----
+
+To create an instance of `HazelcastEmbeddingStore`, use its `builder` method with the dimension of the embedding model:
+
+[source,java]
+----
+var store = HazelcastEmbeddingStore.builder(embeddingModel.dimension())
+    // ...
+    .build();
+----
+
+The `builder` method creates an instance of `HazelcastEmbeddingStore.Builder`.
+
+`HazelcastEmbeddingStore` needs to communicate with a Hazelcast Enterprise cluster to send embeddings and retrieve search results. You can supply cluster configuration parameters using one of the following methods:
```

Comment:
In that case, is this effectively an Enterprise feature and should these new sections be tagged as such?